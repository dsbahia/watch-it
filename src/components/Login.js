import React, { useState } from "react";
import axios from "axios";

function Login() {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [error, setError] = useState(null);

const handleSubmit = async (e) =>




  return <div></div>;
}

export default Login;
