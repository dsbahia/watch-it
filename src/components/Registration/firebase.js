import { initializeApp } from "firebase/app";
import { getAuth } from "firebase/auth";

const firebaseConfig = {
  apiKey: "AIzaSyDpiscXs7UomLcbyhm4Lq8rdp6zUoGiy-s",
  authDomain: "watch-it-be.firebaseapp.com",
  projectId: "watch-it-be",
  storageBucket: "watch-it-be.appspot.com",
  messagingSenderId: "241617376848",
  appId: "1:241617376848:web:5c5ec7579fc6567bed6355",
  measurementId: "G-841E4LK74N",
};
const app = initializeApp(firebaseConfig);
const auth = getAuth(app);
export { auth };
