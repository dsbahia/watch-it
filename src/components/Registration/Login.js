import { React, useState } from "react";
import {
  signInWithEmailAndPassword,
  sendEmailVerification,
} from "firebase/auth";
import { Link, useNavigate } from "react-router-dom";
import { auth } from "./firebase";
import { useAuthValue } from "./AuthContext";
import "../../styles/forms.css";

function Login() {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [error, setError] = useState("");
  const { setTimeActive } = useAuthValue();
  const navigate = useNavigate();

  const login = (e) => {
    e.preventDefault();
    signInWithEmailAndPassword(auth, email, password)
      .then(() => {
        if (!auth.currentUser.emailVerified) {
          sendEmailVerification(auth.currentUser)
            .then(() => {
              setTimeActive(true);
              navigate("/verify-email");
            })
            .catch((err) => alert(err.message));
        } else {
          navigate("/");
        }
      })
      .catch((err) => setError(err.message));
  };

  return (
    <div className="auth-container">
      <div className="auth-form">
        <h1>Log in</h1>
        {error && <div className="auth-error">{error}</div>}
        <form onSubmit={login} name="login_form">
          <input
            type="email"
            value={email}
            required
            placeholder="Enter your email"
            onChange={(e) => setEmail(e.target.value)}
            className="auth-input"
          />

          <input
            type="password"
            value={password}
            required
            placeholder="Enter your password"
            onChange={(e) => setPassword(e.target.value)}
            className="auth-input"
          />

          <button type="submit" className="auth-button">
            Login
          </button>
        </form>
        <span>
          Don&apos;t have an account?
          <Link to="/register">Create one here</Link>
        </span>
      </div>
    </div>
  );
}

export default Login;
