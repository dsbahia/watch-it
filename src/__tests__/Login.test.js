import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import Login from "../components/Registration/Login";

import firebase from "../components/Registration/firebase";

jest.mock("../components/Registration/firebase");

describe("Login Component", () => {
    beforeEach(() => {
        firebase.mockClear();
    });
  it("should render the login form", () => {
    const { getByText, getByPlaceholderText } = render(<Login />);

    // Ensure that the form elements are rendered
    expect(getByText("Log in")).toBeInTheDocument();
    expect(getByPlaceholderText("Enter your email")).toBeInTheDocument();
    expect(getByPlaceholderText("Enter your password")).toBeInTheDocument();
    expect(getByText("Login")).toBeInTheDocument();
    expect(getByText("Don't have an account?")).toBeInTheDocument();
  });

  it("should show an error message on incorrect login credentials", async () => {
    const { getByText, getByPlaceholderText, getByRole } = render(<Login />);
    const emailInput = getByPlaceholderText("Enter your email");
    const passwordInput = getByPlaceholderText("Enter your password");
    const loginButton = getByRole("button", { name: "Login" });

    // Simulate user input
    fireEvent.change(emailInput, { target: { value: "test@example.com" } });
    fireEvent.change(passwordInput, { target: { value: "incorrectpassword" } });

    fireEvent.click(loginButton);

    // Wait for error message to appear
    await waitFor(() => {
      expect(
        getByText("Incorrect email or password. Please try again."),
      ).toBeInTheDocument();
    });
  });

  // You can add more tests for other scenarios here
});
