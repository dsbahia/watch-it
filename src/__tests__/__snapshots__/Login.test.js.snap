// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="auth-container"
  >
    <div
      class="auth-form"
    >
      <h1>
        Log in
      </h1>
      <form
        name="login_form"
      >
        <input
          class="auth-input"
          placeholder="Enter your email"
          required=""
          type="email"
          value=""
        />
        <input
          class="auth-input"
          placeholder="Enter your password"
          required=""
          type="password"
          value=""
        />
        <button
          class="auth-button"
          type="submit"
        >
          Login
        </button>
      </form>
      <span>
        Don't have an account?
        <a
          href="/register"
        >
          Create one here
        </a>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
